client/task_runner.go

```diff
@@ -315,24 +350,23 @@ func (r *TaskRunner) validateTask() error {
 	return mErr.ErrorOrNil()
 }
 
-func (r *TaskRunner) run() {
-	// Predeclare things so we can jump to the RESTART
-	var handleEmpty bool
-	var stopCollection chan struct{}
+// prestart handles life-cycle tasks that occur before the task has started.
+func (r *TaskRunner) prestart(taskDir string) (success bool) {
```

Comment:
TaskDir isn't always initialized, so it is easier to pass in as an arg that will be good